builds/jenkins-build-pack-function/Jenkinsfile

```diff
@@ -982,12 +979,20 @@ def writeServerlessFile(config, env, accountDetails) {
     sh "sed -i -- 's|{event_source_dynamodb}|${event_source_dynamodb}|g' serverless.yml"
   } 
   
-  if (config['domain'] != 'jazz' && (config['event_source_dynamodb'] || config['event_source_sqs'] || config['event_source_kinesis'] || config['event_source_s3'] 
-  || ( configLoader.TVAULT && configLoader.TVAULT.IS_ENABLED ))) {
+  if (config['domain'] != 'jazz' && !config['iamRoleARN'] ) {
```

Comment:
This complete if-else block is little confusing. Can we add comments on when we are doing what?

Reply:
Done